src/PrestaShopBundle/Bridge/AdminController/AdminControllerTrait.php

```diff
@@ -28,114 +28,68 @@
 
 namespace PrestaShopBundle\Bridge\AdminController;
 
-use PrestaShopBundle\Bridge\AdminController\Action\ActionInterface;
-use PrestaShopBundle\Bridge\AdminController\Action\HeaderToolbarAction;
-use PrestaShopBundle\Bridge\AdminController\Action\ListBulkAction;
-use PrestaShopBundle\Bridge\AdminController\Action\ListHeaderToolbarAction;
-use PrestaShopBundle\Bridge\AdminController\Action\ListRowAction;
-use PrestaShopBundle\Bridge\AdminController\Field\FieldInterface;
-use PrestaShopBundle\Bridge\Exception\NotAllowedActionTypeForListException;
-use PrestaShopBundle\Bridge\Exception\NotAllowedGenericActionTypeException;
-use PrestaShopBundle\Bridge\Helper\FiltersHelper;
-use PrestaShopBundle\Bridge\Helper\HelperListBridge;
-use PrestaShopBundle\Bridge\Helper\HelperListConfiguration;
-use PrestaShopBundle\Bridge\Helper\ResetFiltersHelper;
+use PrestaShopBundle\Bridge\Exception\BridgeException;
+use Tab;
 
 /**
- * This trait contains the principal methods you need when you want to migrate a controller horizontally.
- *
- * This trait adds methods to:
- *     - add a general action
- *     - add a specific action to list
- *     - add a field to a controller
- *     - get reset filter helpers
- *     - get filter helpers
- *     - get helper list bridge
+ * Contains reusable methods for horizontally migrated controllers
  */
 trait AdminControllerTrait
 {
     /**
-     * @return ResetFiltersHelper
+     * @var LegacyControllerBridge
      */
-    public function getResetFiltersHelper(): ResetFiltersHelper
-    {
-        return $this->get('prestashop.core.bridge.processor.reset_filters_helper');
-    }
+    private $legacyControllerBridge;
 
     /**
-     * @return FiltersHelper
+     * @return LegacyControllerBridgeInterface
      */
-    public function getFiltersHelper(): FiltersHelper
-    {
-        return $this->get('prestashop.core.bridge.processor.filters_helper');
-    }
-
-    /**
-     * @return HelperListBridge
-     */
-    public function getHelperListBridge(): HelperListBridge
-    {
-        return $this->get('prestashop.core.bridge.helper_list_bridge');
-    }
+    protected function buildControllerBridge(
```

Comment:
The method name is"build", but the method builds and sets the attributes. I think we have to change the method name or separate the responsability

Reply:
It only builds a configuration and injects it into the Bridge, so IMO it is all part of the build process, i don't see any problem here :thinking: